content/blog/pragmatic-architecture.md

```diff
@@ -0,0 +1,61 @@
+
+I'm not a big fan of labelling people, and especially with these kinds of labels I prefer to tread carefully. An Architect coming across as 'somewhat dogmatic' can be a matter of perspective, different communication styles, or it can be a sign that they are, in fact, a bit stuck in their ways.
+
+A good indicator for dogmatic architecture is when the Architectural Principle or the historical evidence has more weight than the actual requirements or constraints for the target architecture. And the Architect might not even be aware it's happening.
+
+## Avoiding dogmatism
+As an experienced Architect, it's hard to not fall victim to dogmatism sometimes. Especially once you think that'll never happen to you.. _especially_ then. It's a careful balancing act between relying on your experience, relying on common and proven architectural patterns, relying on your analysis of the situation at hand, and fact-checking your associative memory. Are you considering something because of the current constraints, or because your experiences primed you to prefer that certain solution?
+
+In my opinion the best starting point is to be very aware of the fact that everyone suffers from tunnel vision. Awareness is half the battle. And while every situation is a little different, there are some things that I try to consistently apply in order to check my biases and my assumptions:
+
+1. At all stages, be clear about what 'your gut feeling' (read: your potential dogmatism) is telling you. Make sure to call it that, too. The information available to you at different stages will lead to different trains of thought. I typically will tell stakeholders that "based on what I know right now, I'm leaning towards XYZ". I may also share my biases if I'm _not_ following them: "Normally, in situations like these I would lean towards XYZ because of \<set of reasons\>, but in this case..." etcetera.
+1. Challenge your gut feeling: "What if I can't solve it this particular way?". It's a thought exercise I regularly do, and a pretty good way to pass time while commuting. But you can even develop it into a competing solution if the situation permits. It also allows me to entertain multiple trains of thought through the discovery process. It takes more time, and often time is at a premium, but if you can do it you should.
+1. Consistently check the priorities and constraints with stakeholders. This helps in managing expectations, adds clarity, and means you're not making too many assumptions here. You can influence priorities as well, by informing your stakeholders on consequences their prioritization may have, which in turn serves your own thought process. Did you consider the impact of certain choices well enough?
```

Comment:
"Did you consider the impact of certain choices well enough?"

Zoals ik het nu lees is dit de realisatie die je bij je stakeholders teweeg wil brengen, correct? Zo komt het bij mij namelijk niet direct over, misschien als je het herschrijft naar:

"Did they consider the impact of certain choices well enough?"

Zo niet, dan is deze zin wellicht wat verwarrend, wellicht dat ie eigenlijk niet eens nodig is.

Reply:
Uiteindelijk dient de discussie met stakeholders ook je eigen gedachtenproces. Het is vrij makkelijk om in je eigen bubbel te kijken naar 'de beste oplossing' en daarbij zelf te prioriteren. Je stakeholders (de business, klant, etc) prioriteert anders, en dat leidt tot bepaalde impact op wat er mogelijk is, of wat er het best werkt. Je moet jezelf voortdurend weten te verplaatsen in de feitelijke context (ipv het ideaalbeeld wat je in je hoofd hebt 😛 )